blog/content/post/2020-03-24-trenchboot-nlnet-introduction.md

```diff
@@ -0,0 +1,107 @@
+contribute these solutions to the public good, not as proprietary products, but
+as free and open source technologies that any person or application can use
+without restrictions and modify to their own needs and the needs of others.
+
+This article is an introduction, which should bring you closer to the overall
+project's concept and describe goals, which we want to achieve. Also, this blog
+post starts a series of articles describing and verifying our progress at work.
+Thanks to that, you can observe development process and always be up-to-date
+about the project.
+
+## Our goal
+
+Our goal is twofold:
+1. Implement DRTM with TrenchBoot for AMD processors.
+2. Create test environment with test suites for users.
+
+Above issues will be described later. Both, as already mentioned, will be fully
+open-source and available for everybody. Solution, which we provide will be
+intended for AMD processors because it doesn't include any closed
+components.
+
+Now, let me briefly discuss each goal, so you can better understand the project.
+Of course, we won't focus on details and all requirements now. Those will be
+systematically updated and presented to community.
+
+#### DRTM with TrenchBoot
+
+TrenchBoot is one of the stages of boot process. It runs directly after firmware
+(coreboot in our case) execution ended. It is a technology which measures and
+verifies running environment. To obtain that, TrenchBoot utilizes **Dynamic Root
+of Trust for Measurements** (DRTM). It is very complex topic and understanding
+it could take too much time. What you need to know (or trust us about) is that
+those measurements are done for given piece of code (given piece of system) and
+are stored in **Trusted Platform Module** (TPM) special registers. It gives
+possibility to automatically verify them against corruption and eventually
+proceed or stop further booting. Also, the user has possibility to check if
+measurements are valid. It can be done from OS level using dedicated tools. So,
+as you can see, TrenchBoot is critical piece of firmware, if you want to make it
+secure. Also, its implementation is fully open-source, so everyone can inspect
+it and contribute.
+
+#### Test environment
+
+All requirements of project are automatically or manually validate after
+development. For this purpose, we need to create entire testing infrastructure
+with suitable hardware and software components. Moreover, all tests will be
+published for users, so they can verify development stage on their own
+platforms.
+
+Final results, most likely at the end of each month, will be presented in form
+of blog articles. Everything should be clearly pointed out and explained. All
+used tools, all carried out procedures, all prepared configurations - everything
+user can inspect, revise, build, reproduce and finally use. The goal is to have
+a user-friendly and trust-worthy environment, so community can be sure about the
+quality of the solution we provide.
+
+## Summary
```

Comment:
@piotr-kleins we have to mention main TrenchBoot repository and other individuals and companies that contribute to the project Oracle, Daniel Smith, Ross, Daniel Kiper, and others. @krystian-hebel cc

Reply:
List of all commiters/reviewers of LZ only:
```
Andrew Cooper
Daniel P. Smith
Krystian Hebel
Michał Żygowski
Piotr Król
Ross Philipson
```
In addition to those there is also one commit from Eric Snowberg to kernel, and of course Daniel Kiper reviews patches send to GRUB.

Reply:
Added in summary section